src/tools/compiletest/src/runtest/ui.rs

```diff
@@ -24,7 +24,19 @@ impl TestCx<'_> {
         let pm = self.pass_mode();
         let should_run = self.should_run(pm);
         let emit_metadata = self.should_emit_metadata(pm);
-        let proc_res = self.compile_test(should_run, emit_metadata);
+        let mut proc_res = self.compile_test(should_run, emit_metadata);
+
+        if self.props.parallel_front_end {
+            // Ensure there is no ice during parallel front end.
+            self.check_no_compiler_crash(&proc_res, false);
+
+            // Repeated testing due to instability in multithreaded environments.
+            for _ in 0..50 {
+                proc_res = self.compile_test(should_run, emit_metadata);
+                self.check_no_compiler_crash(&proc_res, false);
+            }
```

Comment:
Question: this is going to be expensive as running 50 times involves not only spawning multiple processes, but also more expensive is the compiling itself. Do you have any local benchmarks to see how much longer the test will take w/ the extra 49 runs compared to just running once? I'm not suggesting we shouldn't do this -- we *should* -- but I would like to have more info on the impact of this on test time and then make an informed decision to accept the cost.

Context: I'm primarily thinking about some tests that might take a *long* time to compile, e.g. tests that have deep recursive types.

Reply:
I haven't found a good benchmark yet, from what I've observed so far these tests don't noticeably extend the test time though. I think we can keep this comment until we get more information later